File: snow_pkg/src/openalea/SnowDepth.py
# coding: utf8
from copy import copy

import numpy
from math import *

def model_SnowDepth(Snowmelt = 0.0,
         Sdepth_t1 = 0.0,
         Snowaccu = 0.0,
         E = 0.0,
         rho = 100.0):
    """
     - Name: SnowDepth -Version: 1.0, -Time step: 1
     - Description:
                 * Title: snow cover depth Calculation
                 * Author: STICS
                 * Reference: -
                 * Institution: INRA
                 * Abstract: -
     - inputs:
                 * name: Snowmelt
                               ** min : 0.0
                               ** default : 0.0
                               ** max : 100.0
                               ** uri : 
                               ** variablecategory : state
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : m
                               ** description : snow melt 
                 * name: Sdepth_t1
                               ** min : 0.0
                               ** default : 0.0
                               ** max : 5000.0
                               ** uri : 
                               ** variablecategory : state
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : m
                               ** description : snow cover depth Calculation in previous day
                 * name: Snowaccu
                               ** min : 
                               ** default : 0.0
                               ** max : 
                               ** uri : 
                               ** variablecategory : rate
                               ** datatype : DOUBLE
                               ** inputtype : variable
                               ** unit : mmW/d
                               ** description : snowfall accumulation
                 * name: E
                               ** parametercategory : constant
                               ** min : 
                               ** datatype : DOUBLE
                               ** max : 
                               ** uri : 
                               ** default : 0.0
                               ** inputtype : parameter
                               ** unit : mm/mm/d
                               ** description : snow compaction parameter
                 * name: rho
                               ** parametercategory : constant
                               ** min : 
                               ** datatype : DOUBLE
                               ** max : 
                               ** uri : 
                               ** default : 100
                               ** inputtype : parameter
                               ** unit : kg/m**3
                               ** description : The density of the new snow fixed by the user
     - outputs:
                 * name: Sdepth
                               ** min : 0.0
                               ** variablecategory : state
                               ** max : 500.0
                               ** uri : 
                               ** datatype : DOUBLE
                               ** unit : m
                               ** description : snow cover depth Calculation
    """

    Sdepth = 0.0
    if Snowmelt <= Sdepth_t1 + (Snowaccu / rho):
        Sdepth = Snowaccu / rho + Sdepth_t1 - Snowmelt - (Sdepth_t1 * E)
    return Sdepth

File: snow_pkg/src/f90/SnowDepth.f90
MODULE Snowdepthmod
    IMPLICIT NONE
CONTAINS

    SUBROUTINE model_SnowDepth(Snowmelt, &
        Sdepth_t1, &
        Snowaccu, &
        E, &
        rho, &
        Sdepth)
        IMPLICIT NONE
        REAL, INTENT(IN) :: Snowmelt
        REAL, INTENT(IN) :: Sdepth_t1
        REAL, INTENT(IN) :: Snowaccu
        REAL, INTENT(IN) :: E
        REAL, INTENT(IN) :: rho
        REAL, INTENT(OUT) :: Sdepth
        !- Description:
    !            * Title: snow cover depth Calculation
    !            * Author: STICS
    !            * Reference: -
    !            * Institution: INRA
    !            * Abstract: -
        !- inputs:
    !            * name: Snowmelt
    !                          ** description : snow melt 
    !                          ** inputtype : variable
    !                          ** variablecategory : state
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 100.0
    !                          ** unit : m
    !                          ** uri : 
    !            * name: Sdepth_t1
    !                          ** description : snow cover depth Calculation in previous day
    !                          ** inputtype : variable
    !                          ** variablecategory : state
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 0.0
    !                          ** max : 5000.0
    !                          ** unit : m
    !                          ** uri : 
    !            * name: Snowaccu
    !                          ** description : snowfall accumulation
    !                          ** inputtype : variable
    !                          ** variablecategory : rate
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 
    !                          ** max : 
    !                          ** unit : mmW/d
    !                          ** uri : 
    !            * name: E
    !                          ** description : snow compaction parameter
    !                          ** inputtype : parameter
    !                          ** parametercategory : constant
    !                          ** datatype : DOUBLE
    !                          ** default : 0.0
    !                          ** min : 
    !                          ** max : 
    !                          ** unit : mm/mm/d
    !                          ** uri : 
    !            * name: rho
    !                          ** description : The density of the new snow fixed by the user
    !                          ** inputtype : parameter
    !                          ** parametercategory : constant
    !                          ** datatype : DOUBLE
    !                          ** default : 100
    !                          ** min : 
    !                          ** max : 
    !                          ** unit : kg/m**3
    !                          ** uri : 
        !- outputs:
    !            * name: Sdepth
    !                          ** description : snow cover depth Calculation
    !                          ** variablecategory : state
    !                          ** datatype : DOUBLE
    !                          ** min : 0.0
    !                          ** max : 500.0
    !                          ** unit : m
    !                          ** uri : 
        Sdepth = 0.0
        IF(Snowmelt .LE. Sdepth_t1 + (Snowaccu / rho)) THEN
            Sdepth = Snowaccu / rho + Sdepth_t1 - Snowmelt - (Sdepth_t1 * E)
        END IF
    END SUBROUTINE model_SnowDepth

END MODULE
